src/components/PeoplePage.tsx

```diff
@@ -1,29 +1,162 @@
+/* eslint-disable react-hooks/exhaustive-deps */
+/* eslint-disable max-len */
 import { PeopleFilters } from './PeopleFilters';
 import { Loader } from './Loader';
 import { PeopleTable } from './PeopleTable';
+import { useEffect, useMemo, useState } from 'react';
+import { Person } from '../types';
+import { getPeople } from '../api';
+import { useSearchParams } from 'react-router-dom';
 
 export const PeoplePage = () => {
+  const [people, setPeople] = useState<Person[]>([]);
+  const [loading, setLoading] = useState(false);
+  const [error, setError] = useState('');
+  const [searchParams, setSearchParams] = useSearchParams();
+  const query = searchParams.get('query') || '';
+  const sex = searchParams.get('sex') || 'all';
+  const centuries = searchParams.getAll('centuries') || [];
+
+  function filter() {
+    const preparedPeople = [...people];
+
+    const queryFiltered = preparedPeople.filter(person =>
+      person.name?.toLowerCase().includes(query),
+    );
+
+    if (sex === 'all') {
+      return queryFiltered;
+    } else {
+      return queryFiltered.filter(person => person.sex === sex);
+    }
+  }
+
+  const filteredPeople = useMemo(() => {
+    return filter();
+  }, [sex, people, query]);
+
+  const sexFilter = (value: string) => {
+    const params = new URLSearchParams(searchParams);
+
+    params.set('sex', value);
+    setSearchParams(params);
+  };
+
+  const queryFilter = (e: React.ChangeEvent<HTMLInputElement>) => {
+    const params = new URLSearchParams(searchParams);
+
+    params.set('query', e.target.value);
+    setSearchParams(params);
+  };
+
+  const chooseCenturies = (n: string) =>
+    centuries.includes(n)
+      ? centuries.filter(century => century !== n)
+      : [...centuries, n];
```

Comment:
The `chooseCenturies` function does not update the search parameters correctly. It should modify the `searchParams` to reflect the changes in selected centuries, similar to how `sexFilter` and `queryFilter` update the search parameters.